components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/wsdl/WSDL11SOAPOperationExtractor.java

```diff
@@ -386,11 +388,55 @@ private void addModelDefinition(Node current, ModelImpl model, String type, bool
+                                    model.getProperties().remove(element);
+                                    parentProp = createPropertyFromNode(current, true);
+                                    parentProp.setName(element);
+                                    model.addProperty(element, parentProp);
+                                } else {
+                                    if (isChildNode(current, prevNode)) {
+                                        int index = Arrays.asList(elements).indexOf(element);
+                                        Property parentProperty = null;
+                                        String parentName = "";
+                                        for (int i = 1; i < index; i++) {
+                                            parentProperty = model.getProperties().get(elements[index - i]);
+                                            if (parentProperty != null) {
+                                                model.getProperties().remove(element);
+                                                parentName = elements[index - i];
+                                                break;
+                                            }
+                                        }
+                                        if (parentProperty instanceof ArrayProperty) {
+                                            if (((ArrayProperty) parentProperty).getItems() != null) {
+                                                if (parentName.equals(((ArrayProperty) parentProperty).getItems()
+                                                        .getName())) {
+                                                    Property objectProperty = new ObjectProperty();
+                                                    Map propertiesMap = new HashMap();
+                                                    propertiesMap.put(element, createPropertyFromNode(current, true));
+                                                    ((ObjectProperty) objectProperty).setProperties(propertiesMap);
+                                                    ((ArrayProperty) parentProperty).setItems(objectProperty);
+                                                    parentProp = parentProperty;
+                                                }
+                                            } else {
+                                                ((ArrayProperty) parentProperty)
+                                                        .setItems(createPropertyFromNode(current, true));
+                                            }
+                                        } else if (parentProperty instanceof ObjectProperty) {
+                                            if (((ObjectProperty) parentProperty).getProperties() != null) {
+                                                if ((((ObjectProperty) parentProperty).getProperties()).get(parentName) != null) {
```

Comment:
Please check whether the length is exceeded?

Reply:
Fixed